buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt

```diff
@@ -244,6 +245,44 @@ class PerformanceTestPlugin : Plugin<Project> {
             channel = "adhoc"
             outputs.doNotCacheIf("Is adhoc performance test") { true }
         }
+
+        val shouldLoadScenariosFromFile = project.providers.gradleProperty("includePerformanceTestScenarios")
+            .forUseAtConfigurationTime()
+            .getOrElse("false") == "true"
+
+        configureSampleGenerators {
+            all {
+                val sampleGenerator = this
```

Comment:
Why don't we do something like this instead?

```suggestion
            all { sampleGenerator ->
```

Reply:
This would call the wrong `all` method - the one from Kotlin returning a `boolean`.

Reply:
This is very confusing. :(

Reply:
It is, and I got it wrong and didn't understand what is going on.

Reply:
@eskatos @bamboo can you guys give us a hand here? How should this code be written to reduce confusion? Thanks!

Reply:
Either you capture `this` in an explicit variable as done above, or you use the implicit receiver directly. There isn't more to it.